build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/buildmodel/BuildDescription.java

```diff
@@ -1597,7 +1599,8 @@ private void calculateInputs(BuildStep step) throws CoreException {
 										if ((var = var.trim()).length() != 0) {
 											if (VAR_USER_OBJS.equals(var)) {
 												String objs[] = getUserObjs(step);
-												if (objs != null && objs.length != 0) {
+												// if there are user objects and they will not be provided as extra flags
+												if (objs != null && objs.length != 0 && !toolHasExtraFlags) {
```

Comment:
I would have put `!toolHasExtraFlags` first here since it's cheaper to do that check than the rest.

Reply:
Done.